tools/export/iar/__init__.py

```diff
@@ -111,6 +111,8 @@ def generate(self):
                self.resources.c_sources + self.resources.cpp_sources + \
                self.resources.objects + self.resources.libraries
         flags = self.flags
+        _, macros = self.toolchain.config.get_config_data()
+        defines = ['-D%s' % d for d in macros] if macros else [""]
```

Comment:
Could you do:
```python
macros = self.toolchain.get_symbols(True)
defines = self.toolchain.get_compile_options(macros, self.resources.inc_dirs, True)
```
so that it exactly matches the code run in `iar.assemble`?

Reply:
Same thing here, **defines** does not include the macros from mbed_app.json and mbed_lib.json in **macros** with this.